torchao/float8/inference.py

```diff
@@ -24,7 +25,27 @@
     ScaledMMConfig,
     tensor_already_casted_to_fp8,
 )
-from torchao.float8.float8_utils import e4m3_dtype, tensor_to_scale
+from torchao.float8.float8_utils import (
+    e4m3_dtype,
+    tensor_to_scale,
+    get_rowwise_tile_size,
+    flatten_input,
+    get_out_shape,
+)
+from torchao.float8.float8_python_api import addmm_float8_unwrapped
+
+
+class ScalingGranularity(Enum):
+    """Granularity of the scaling factor
+
+    GLOBAL: The same scale is used for all elements
+    PER_CHANNEL: A different scale is used for each channel
+    PER_TENSOR: A different scale is used for each tensor
+    """
+
+    TENSOR_WISE = auto()
```

Comment:
nit, can we use strings, so it's easier to convert from string to enum? this is useful to avoid additional boilerplate when working with commandline scripts which take these things as user input

Reply:
will do

Reply:
once we are 3.11 or greater would be nice to use: https://docs.python.org/3/library/enum.html#enum.StrEnum